### PR TITLE
Current progress

### PR DESCRIPTION
**Pull request by msarvar**

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -773,7 +786,7 @@ func (s *Server) waitForDrain() {
 }
 
 func (s *Server) LogStreaming(w http.ResponseWriter, _ *http.Request) {
```
Comment (msarvar) [inline server/server.go:788]: You can delete this function completely.

Comment (isatasan) [reply server/server.go:788]: 👍

File: server/controllers/templates/web_templates.go [hunk 2]
```diff
@@ -352,6 +352,75 @@ v{{ .AtlantisVersion }}
...
+    </footer>
+
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script>
+    <script src="https://unpkg.com/xterm@4.9.0/lib/xterm.js"></script>
+    <script src="https://unpkg.com/xterm-addon-attach@0.6.0/lib/xterm-addon-attach.js"></script>
+    <script src="https://unpkg.com/xterm-addon-fit@0.4.0/lib/xterm-addon-fit.js"></script>
+
+    <script>
+      var term = new Terminal();
+      var socket = new WebSocket("wss://log_streaming_ws");
```
Comment (msarvar) [inline server/controllers/templates/web_templates.go:411]: This needs to be full url to the server. Upstream [example](https://github.com/runatlantis/atlantis/pull/1315/files#diff-98446d84bdd9f6860dfe6c4d3f011675dbe05898a57034cd04e84a2489793175R292). Something like this
```suggestion
      var socket = new WebSocket(
      (document.location.protocol === "http:" ? "ws://" : "wss://") +
         document.location.host +
         "/log_streaming_ws");
```

File: server/controllers/templates/web_templates.go [hunk 3]
```diff
@@ -352,6 +352,75 @@ v{{ .AtlantisVersion }}
...
+    </footer>
+
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script>
+    <script src="https://unpkg.com/xterm@4.9.0/lib/xterm.js"></script>
+    <script src="https://unpkg.com/xterm-addon-attach@0.6.0/lib/xterm-addon-attach.js"></script>
+    <script src="https://unpkg.com/xterm-addon-fit@0.4.0/lib/xterm-addon-fit.js"></script>
+
+    <script>
+      var term = new Terminal();
+      var socket = new WebSocket(document.location.protocol === "http:" ? "ws://" : "wss://") + document.location.host + "/ws");
```
Comment (msarvar) [inline server/controllers/templates/web_templates.go:411]:
```suggestion
      var socket = new WebSocket(
      (document.location.protocol === "http:" ? "ws://" : "wss://") + 
      document.location.host + 
      document.location.pathname +
      "/ws");
```

File: server/web_templates.go [hunk 4]
```diff
@@ -0,0 +1,447 @@
+// Copyright 2017 HootSuite Media Inc.
```
Comment (msarvar) [inline server/web_templates.go:1]: This file can be removed

**Comment by msarvar** [comment]
can you address some of the linting issues in https://github.com/lyft/atlantis/pull/86/checks?check_run_id=3060674551? I can merge it after that.

File: server/events/models/models.go [hunk 5]
```diff
@@ -435,6 +435,11 @@ func (p ProjectCommandContext) GetShowResultFileName() string {
 	return fmt.Sprintf("%s-%s.json", projName, p.Workspace)
 }
 
+// Gets a unique identifier for the current pull request as a single string
+func (p *ProjectCommandContext) PullInfo() string {
```
Comment (msarvar) [inline server/events/models/models.go:439]: To keep things consistent let make this change.
```suggestion
func (p ProjectCommandContext) PullInfo() string {
```

File: server/controllers/templates/web_templates.go [hunk 6]
```diff
@@ -116,7 +116,7 @@ var IndexTemplate = template.Must(template.New("index.html.tmpl").Parse(`
         <div class="list-title">{{.RepoFullName}} <span class="heading-font-size">#{{.PullNum}}</span> <code>{{.Path}}</code> <code>{{.Workspace}}</code></div>
         <div class="list-status"><code>Locked</code></div>
         <div class="list-timestamp"><span class="heading-font-size">{{.TimeFormatted}}</span></div>
-        </div>
+        </div>x
```
Comment (msarvar) [inline server/controllers/templates/web_templates.go:119]:
```suggestion
        </div>
```

**Review by nishkrishnan — APPROVED** [review]

**Pull request merged**